src/transaction/log_page_buffer.c

```
@@ -7303,6 +7304,97 @@ logpb_checkpoint (THREAD_ENTRY * thread_p)
+      }
+    LOG_LSA dummy_smallest_tran_lsa = NULL_LSA;
+    trantable_checkpoint_info.load_trantable_snapshot (thread_p, dummy_smallest_tran_lsa);
+
+    // loading the transaction table snapshot ensures also that a snapshot lsa has been set
+    const log_lsa trantable_checkpoint_lsa = trantable_checkpoint_info.get_snapshot_lsa ();
+
+    if (trantable_checkpoint_lsa.is_null ())
+      {
+	// no transaction table checkpoint can be recorded
+	er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: failed; null checkpoint lsa\n");
+	return ER_FAILED;
+      }
+
+    if (detailed_logging)
+      {
+	_er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: adding with lsa=%lld|%d\n",
+		       LSA_AS_ARGS (&trantable_checkpoint_lsa));
+      }
+    log_Gl.m_metainfo.add_checkpoint_info (trantable_checkpoint_lsa, std::move (trantable_checkpoint_info));
+
+    // make sure new checkpoint is persisted to disk
+    const int res_metalog_to_file = log_write_metalog_to_file ();
+    if (res_metalog_to_file != NO_ERROR)
+      {
+	ASSERT_ERROR ();
+	er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: failed; writing metalog to file\n");
+	return res_metalog_to_file;
+      }
+
+    logpb_flush_pages (thread_p, &trantable_checkpoint_lsa);
+
+    // drop previous checkpoints
+    if (detailed_logging)
+      {
+	_er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: droping previous before lsa=%lld|%d\n",
+		       LSA_AS_ARGS (&trantable_checkpoint_lsa));
+      }
+    log_Gl.m_metainfo.remove_checkpoint_info_before_lsa (trantable_checkpoint_lsa);
+
+    // - in nominal conditions, there should be at most one previous trantable checkpoint
+    // - in abnormal conditions (such as when the system crashed just after adding a new trantable
+    //    checkpoint and before deleting the outdated checkpoint) there can be at most two
+    assert (log_Gl.m_metainfo.get_checkpoint_count () == 1);
+
+    // make sure new checkpoint is persisted to disk
+    log_write_metalog_to_file ();
```

If error is ignored intentionally, add `(void)` to show the intention.
```suggestion
    (void) log_write_metalog_to_file ();
```